controller/scan/registry.go

```diff
@@ -1544,29 +1550,21 @@ func (rs *Registry) scheduleScanImages(
 					scanTypesRequired.Vulnerability = true
 				}
 
-				signatureInfoChanged := info.SignatureDigest != sum.SignatureDigest
-				sigstoreConfigurationChanged := sigstoreTimestamp != sum.SigstoreTimestamp
-
-				if signatureInfoChanged {
+				// determine if signature scan is required
+				if !newImage {
+					scanTypesRequired.Signature = false
+				} else if signatureInfoChanged {
+					smd.scanLog.WithFields(log.Fields{"registry": rs.config.Name, "image": image, "status": sum.Status, "imageID": info.ID}).Debug("Signature info for image changed, signature scan required.")
+					scanTypesRequired.Signature = true
 					sum.SignatureDigest = info.SignatureDigest
 					imageChanged = true
+				} else if sigstoreConfigurationChanged && isSignedImage {
+					smd.scanLog.WithFields(log.Fields{"registry": rs.config.Name, "image": image, "status": sum.Status, "imageID": info.ID}).Debug("Sigstore config changed and image is signed, signature scan required.")
 					scanTypesRequired.Signature = true
-					smd.scanLog.WithFields(log.Fields{"registry": rs.config.Name, "image": image, "status": sum.Status, "imageID": info.ID}).Debug("Signature Info Changed")
-				}
-
-				if sigstoreConfigurationChanged {
 					sum.SigstoreTimestamp = sigstoreTimestamp
```

Comment:
When newImage is true & imageChanged is true & signatureInfoChanged is false and (sigstoreConfigurationChanged && isSignedImage) is true, it reaches here.
sum.SigstoreTimestamp is set to the latest sigstore timestamp.
Then in line 1594 the summary is written to kv and scan task is scheduled.
Does it mean if the scan fails later(because of network, registry, ....), sum.SigstoreTimestamp is already set to the latest sigstore timestamp? Is this correct?

Reply:
Similar situation to `signatureDigest`, see this comment:
https://github.com/neuvector/neuvector/pull/1034#discussion_r1339259333